### DISCUSSION TITLE
Question about CombinerGenerator

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#24763] [Q&A General] [answered]
Hi all,

I am loading in two meshes and combining them into a disjoint mesh using combiner generator using the method showed below.
```
[Mesh]
  [refine]
    type = FileMeshGenerator
    file = '../../gold/ref_sec.msh'
  []
  [coarse]
    type = FileMeshGenerator
    file = '../../gold/coarse_sec2.msh'
  []
  [combine]
    type = CombinerGenerator
    inputs = 'coarse refine'
  []
[]
```
My issue is that only the subdomain ID's on the 'coarse' mesh are being preserved. The subdomains on the 'refine' mesh are being discarded by CombinerGenerator. It seems that which ever mesh is listed first in 'inputs' has its subdomain IDs preserved. I need the subdomain IDs on both meshes to be preserved. Is there a workaround for this?

edit: Note that all of the subdomainIDs between both meshes are unique

**Comment by GiudGiud** [comment, marked as answer]
Hello

Not currently. This is a known issue, scheduled to be addressed over the summer. 
For now you can either modify the generator yourself, or combine the meshes with external software first. 

Guillaume

**Reply by GiudGiud** [reply]
We addressed sideset IDs there #20825

For domain ids & names:
Tied to https://github.com/idaholab/moose/discussions/24577

**Reply by TheGreatCid** [reply]
Got it, thanks!

**Comment by MengnanLi91** [comment]
The issue is that these two meshes doesn't have a proper block name. Can you try adding below mesh generator for each mesh and then combine? This should preserve both subdomain names
```
[refine_block]
   type = RenameBlockGenerator
   input = refine
   old_block = /*subdomainID here*/
   new_block = /*block name*/
 []
 ```
